gitlab/resource_gitlab_instance_cluster_test.go

```diff
@@ -0,0 +1,301 @@
+	Name                        string
+	Domain                      string
+	EnvironmentScope            string
+	KubernetesApiURL            string
+	KubernetesCACert            string
+	KubernetesAuthorizationType string
+	ManagementProjectID         string
+}
+
+func testAccCheckGitlabInstanceClusterExists(n string, cluster *gitlab.InstanceCluster) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %q", n)
+		}
+
+		instanceClusterID, err := instanceClusterIdFromId(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		conn := testAccProvider.Meta().(*gitlab.Client)
+
+		gotCluster, _, err := conn.InstanceCluster.GetCluster(instanceClusterID)
+		if err != nil {
+			return err
+		}
+
+		*cluster = *gotCluster
+
+		return nil
+	}
+}
+
+func testAccCheckGitlabInstanceClusterDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*gitlab.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "gitlab_instance_cluster" {
+			continue
+		}
+
+		instanceClusterID, err := instanceClusterIdFromId(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		gotCluster, resp, err := conn.InstanceCluster.GetCluster(instanceClusterID)
+		fmt.Println(gotCluster)
+		if err == nil {
+			if gotCluster != nil && gotCluster.ID == instanceClusterID {
+				return fmt.Errorf("instance cluster still exists")
+			}
+		}
+		if resp.StatusCode != 404 {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckGitlabInstanceClusterAttributes(cluster *gitlab.InstanceCluster, want *testAccGitlabInstanceClusterExpectedAttributes) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if cluster.Name != want.Name {
+			return fmt.Errorf("got name %q; want %q", cluster.Name, want.Name)
+		}
+
+		if cluster.Domain != want.Domain {
+			return fmt.Errorf("got domain %q; want %q", cluster.Domain, want.Domain)
+		}
+
+		if cluster.EnvironmentScope != want.EnvironmentScope {
+			return fmt.Errorf("got environment scope %q; want %q", cluster.EnvironmentScope, want.EnvironmentScope)
+		}
+
+		if cluster.PlatformKubernetes.APIURL != want.KubernetesApiURL {
+			return fmt.Errorf("got kubernetes api url %q; want %q", cluster.PlatformKubernetes.APIURL, want.KubernetesApiURL)
+		}
+
+		if cluster.PlatformKubernetes.CaCert != want.KubernetesCACert {
+			return fmt.Errorf("got kubernetes ca cert %q; want %q", cluster.PlatformKubernetes.CaCert, want.KubernetesCACert)
+		}
+
+		if cluster.PlatformKubernetes.AuthorizationType != want.KubernetesAuthorizationType {
+			return fmt.Errorf("got kubernetes authorization type %q; want %q", cluster.PlatformKubernetes.AuthorizationType, want.KubernetesAuthorizationType)
+		}
+
+		return nil
+	}
+}
+
+func testAccGitlabInstanceClusterConfig(rInt int, managed bool) string {
+	m := fmt.Sprintf("%t", managed)
+	if managed {
+		m = "true"
+	}
+
+	return fmt.Sprintf(`
+variable "cert" {
```

Comment:
IMO the test would be clearer if you set the `kubernetes_ca_cert` attribute inline instead of setting it indirectly with a TF variable. Would this work?

```hcl
kubernetes_ca_cert = <<EOF
%s
EOF
```

Reply:
Yeah it's ok how it is. Sorry for the nitpick! 😓